### PR TITLE
Fix loader load subAsset with the same url issue

### PR DESCRIPTION
**Pull request by pandamicro**
Re: cocos-creator/fireball#4941

Changes proposed in this pull request:
 * Fix loader load subAsset with the same url issue

> Makes sure these boxes are checked before submitting your PR - thank you!
>
- [x] If your pull request has gone "stale", you should **rebase** your work on top of the latest version of the upstream branch.
- [x] If your commit history is full of small, unimportant commits (such as "fix pep8" or "update tests"), **squash** your commits down to a few, or one, discreet changesets before submitting a pull request.
- To official teams:
  - [x] Check that your javascript is following our [style guide](https://github.com/cocos-creator/fireball/blob/dev/.github/CONTRIBUTING.md) and end files with a newline
  - [x] Document new code with comments in source code based on [API Docs](https://github.com/cocos-creator/fireball#api-docs)
  - [x] Make sure any log information in `cc.log` , `cc.error`, `cc.warn` or `cc.assert` has been moved into `DebugInfos.js` with an ID

@cocos-creator/engine-admins

### CODE REVIEW

File: cocos2d/core/load-pipeline/CCLoader.js [hunk 1]
```diff
@@ -55,9 +57,10 @@ function getResWithUrl (res) {
         result = {};
         id = res;
     }
-    url = cc.AssetLibrary._getAssetUrl(id);
-    result.url = url || id;
-    if (url && result.type === 'uuid' && cc.AssetLibrary._isRawAsset(id)) {
+    isAsset = cc.AssetLibrary._getAssetUrl(id);
```
Comment (jareguo) [inline cocos2d/core/load-pipeline/CCLoader.js]: 这个变量建议叫做 isUuid

File: cocos2d/core/load-pipeline/CCLoader.js [hunk 2]
```diff
@@ -55,9 +57,10 @@ function getResWithUrl (res) {
...
+    cc.AssetLibrary._getAssetInfoInRuntime(id, _info);
```
Comment (jareguo) [inline cocos2d/core/load-pipeline/CCLoader.js:61]: 这个 API 要运行时才会有效，编辑器下如果也要执行同样操作就要改用异步的 queryAssetInfo

File: cocos2d/core/load-pipeline/CCLoader.js [hunk 3]
```diff
@@ -55,9 +57,10 @@ function getResWithUrl (res) {
         result = {};
         id = res;
     }
-    url = cc.AssetLibrary._getAssetUrl(id);
-    result.url = url || id;
-    if (url && result.type === 'uuid' && cc.AssetLibrary._isRawAsset(id)) {
```
Comment (jareguo) [inline cocos2d/core/load-pipeline/CCLoader.js:60]: _isRawAsset 这个 API 可以去掉了吧？

**Review by jareguo — CHANGES_REQUESTED** [review]

File: cocos2d/core/load-pipeline/CCLoader.js (hunk 2 above)
Comment (pandamicro) [reply cocos2d/core/load-pipeline/CCLoader.js:61]: 没关系，编辑器下面获取不到，isUuid 为 false，就会直接传 uuid 了，然后会走 downloadUuid

File: cocos2d/core/load-pipeline/CCLoader.js (hunk 3 above)
Comment (pandamicro) [reply cocos2d/core/load-pipeline/CCLoader.js:60]: 已更新

**Pull request merged**